internal/runners/show/show.go

```diff
@@ -165,3 +198,108 @@ func scriptsData(project *projectfile.Project, conditional *constraints.Conditio
 
 	return data, nil
 }
+
+func platformsData(owner, project string, branchID strfmt.UUID) ([]string, error) {
+	remotePlatforms, fail := model.FetchPlatformsForCommit(branchID)
+	if fail != nil {
+		return nil, locale.WrapError(fail.ToError(), "err_show_get_platforms", "Could not get platform details for commit: {{.V0}}", branchID.String())
+	}
+
+	platforms := make([]string, len(remotePlatforms))
+	for i, plat := range remotePlatforms {
+		if plat.Kernel == nil || plat.Kernel.Name == nil {
+			continue
+		}
+		platforms[i] = *plat.Kernel.Name
+	}
+
+	return platforms, nil
+}
+
+func languagesData(owner, project string) ([]string, error) {
+	platformLanguages, fail := model.FetchLanguagesForProject(owner, project)
+	if fail != nil {
+		return nil, locale.WrapError(fail.ToError(), "err_show_get_langauges", "Could not get languages for project")
+	}
+
+	languages := make([]string, len(platformLanguages))
+	for i, pl := range platformLanguages {
+		languages[i] = fmt.Sprintf("%s-%s", pl.Name, pl.Version)
+	}
+
+	return languages, nil
+}
+
+func visibilityData(owner, project string, remoteProject *mono_models.Project) string {
+	if remoteProject.Private {
+		return "Private"
+	}
+	return "Public"
+}
+
+func commitsData(owner, project string, commitID strfmt.UUID, localProject *project.Project) (string, error) {
+	latestCommit, fail := model.LatestCommitID(owner, project)
+	if fail != nil {
+		return "", locale.WrapError(fail.ToError(), "err_show_get_latest_commit", "Could not get latest commit ID")
+	}
+
+	if !authentication.Get().Authenticated() {
+		return latestCommit.String(), nil
+	}
+
+	if localProject != nil && localProject.Owner() == owner && localProject.Name() == project {
+		behind, fail := model.CommitsBehindLatest(owner, project, localProject.CommitID())
+		if fail != nil {
+			return "", locale.WrapError(fail.ToError(), "err_show_commits_behind", "Could not determine number of commits behind latest")
+		}
+		if behind != 0 {
+			return fmt.Sprintf("%s (%d behind latest)", latestCommit, behind), nil
+		}
+	}
+
+	return latestCommit.String(), nil
+}
+
+func secretsData(owner, project string) (map[string][]string, error) {
```

Comment:
Please use a typed struct, so something like:

```
type secretOutput {
  User []string
  Project []string
}
```

Reply:
This is what I originally had, however it led to a tabbing issue when outputting the show information, hence the `map[string][]string`. I've converted it back. Here is an example of the output now:

```
Secrets:
User:
 - foox
 - BLAH
 - world
 - FOO
 - foo3
 - TooManyPizza
 - DOESTHISSHOWUP
Project:
 - foo
```

I spent some time in the output library but couldn't find a simple solution to fix this. If you know of a quick fix, or have another way you would like this presented let me know. Otherwise I will create a ticket to update the output library for this use case